rust/connlib/tunnel/src/gateway.rs

```diff
@@ -311,23 +288,25 @@ impl GatewayState {
     pub fn handle_domain_resolved(
         &mut self,
         req: ResolveDnsRequest,
-        addresses: Vec<IpAddr>,
+        resolve_result: Result<Vec<IpAddr>>,
         now: Instant,
     ) -> anyhow::Result<()> {
         use p2p_control::dns_resource_nat;
 
-        let nat_status = self
-            .peers
-            .get_mut(&req.client)
-            .context("Unknown peer")?
-            .setup_nat(
-                req.domain.clone(),
-                req.resource,
-                &addresses,
-                req.proxy_ips,
-                now,
-            )
-            .map(|()| dns_resource_nat::NatStatus::Active)
+        let nat_status = resolve_result
+            .and_then(|addresses| {
+                self.peers
+                    .get_mut(&req.client)
+                    .context("Unknown peer")?
```

Comment:
Is it worth logging a better error here? I seem to remember seeing a few of these before, but I could very well be mistaking it for a different error.

Reply:
If this happens, then the peer disconnected (and we hit an ICE timeout) whilst we were resolving a domain for it. Pretty unlikely to happen I'd say because it would mean that DNS resolution took longer than the ICE timeout (given that we received the domain name via the tunnel). It may happen in other code paths though but there isn't much we can do unless we start to remember, which peers we used to be connected to.